lib/response.js

```diff
@@ -46,7 +46,18 @@ function Response (req, onEnd, reject) {
 
 util.inherits(Response, http.ServerResponse)
 
+let timeoutHandle
+Response.prototype.setTimeout = function (msecs, callback) {
+  if (callback) {
+    timeoutHandle = setTimeout(callback, msecs)
```

Comment:
We will need to emit the `'timeout'` event when the setTimeout triggers (independently of the presence of the callback)

See https://nodejs.org/api/net.html#net_socket_settimeout_timeout_callback for more details.

Reply:
@mcollina according to the docs, the timeout event should be emitted by the socket. how can I cet a reference to the underlying socket?

Reply:
is calling `this.socket.emit('timeout')` enough?